ee/codegen/src/__test__/workflow-sandbox.test.ts

```diff
@@ -580,5 +580,84 @@ describe("Workflow Sandbox", () => {
       // THEN it should generate MockNodeExecution with empty Outputs()
       expect(result).toMatchSnapshot();
     });
+
+    it("should handle mocks with when_condition referencing non-existent node ID", async () => {
+      /**
+       * Tests that mocks with when_condition containing NODE_OUTPUT reference
+       * to a non-existent node ID are handled gracefully by generating None
+       * for the unresolved reference.
+       */
+
+      const writer = new Writer();
+      const uniqueWorkflowContext = workflowContextFactory({ strict: false });
+      const inputVariable: VellumVariable = {
+        id: "1",
+        key: "test_input",
+        type: "STRING",
+      };
+
+      uniqueWorkflowContext.addInputVariableContext(
+        inputVariableContextFactory({
+          inputVariableData: inputVariable,
+          workflowContext: uniqueWorkflowContext,
+        })
+      );
+
+      const genericNodeData = genericNodeFactory();
+      await nodeContextFactory({
+        workflowContext: uniqueWorkflowContext,
+        nodeData: genericNodeData,
+      });
+
+      // GIVEN a mock with when_condition referencing a non-existent node ID
+      const sandboxInputs: WorkflowSandboxDatasetRow[] = [
+        {
+          label: "Scenario with non-existent node in when_condition",
+          inputs: [
+            {
+              name: inputVariable.key,
+              type: "STRING",
+              value: "test-value",
+            },
+          ],
+          mocks: [
+            {
+              node_id: genericNodeData.id,
+              when_condition: {
+                type: "BINARY_EXPRESSION",
+                operator: "=",
+                lhs: {
+                  type: "NODE_OUTPUT",
```

Comment:
DevinAI Let's try an execution counter here in order to reproduce the bug we're seeing